plenum/test/common/test_config_util.py

```diff
@@ -0,0 +1,130 @@
+                                required=False)
+    assert default_config.int_val == 1
+    assert default_config.str_val == 'value1'
+
+
+def test_extend_with_external_config_not_req_not_exist(default_config, default_config_file_system):
+    assert default_config.int_val == -1
+    assert default_config.str_val == 'value-1'
+    extend_with_external_config(default_config, ("/some/bad/path", "some_bad.py"), required=False)
+    assert default_config.int_val == -1
+    assert default_config.str_val == 'value-1'
+
+
+def test_extend_with_default_external_config_no_any(default_config, default_config_file_system):
+    gen_net_dir = os.path.join(default_config.GENERAL_CONFIG_DIR, default_config.NETWORK_NAME)
+    gen_base_dir = os.path.join(default_config.baseDir, default_config.NETWORK_NAME)
+    gen_cfg_file = os.path.join(default_config.GENERAL_CONFIG_DIR, default_config.GENERAL_CONFIG_FILE)
+    net_cfg_file = os.path.join(gen_net_dir, default_config.NETWORK_CONFIG_FILE)
+    usr_cfg_file = os.path.join(gen_base_dir, default_config.USER_CONFIG_FILE)
+    os.remove(gen_cfg_file)
+    os.remove(net_cfg_file)
+    os.remove(usr_cfg_file)
+    assert default_config.int_val == -1
+    assert default_config.str_val == 'value-1'
+    extend_with_default_external_config(default_config)
+    assert default_config.int_val == -1
+    assert default_config.str_val == 'value-1'
+
+
+def test_extend_with_default_external_config_only_gen(default_config, default_config_file_system):
+    gen_net_dir = os.path.join(default_config.GENERAL_CONFIG_DIR, default_config.NETWORK_NAME)
+    gen_base_dir = os.path.join(default_config.baseDir, default_config.NETWORK_NAME)
+    net_cfg_file = os.path.join(gen_net_dir, default_config.NETWORK_CONFIG_FILE)
+    usr_cfg_file = os.path.join(gen_base_dir, default_config.USER_CONFIG_FILE)
+    os.remove(net_cfg_file)
+    os.remove(usr_cfg_file)
+    assert default_config.int_val == -1
+    assert default_config.str_val == 'value-1'
+    extend_with_default_external_config(default_config)
+    assert default_config.int_val == 1
+    assert default_config.str_val == 'value1'
+
+
+def test_extend_with_default_external_config_gen_net(default_config, default_config_file_system):
+    gen_base_dir = os.path.join(default_config.baseDir, default_config.NETWORK_NAME)
+    usr_cfg_file = os.path.join(gen_base_dir, default_config.USER_CONFIG_FILE)
+    os.remove(usr_cfg_file)
+    assert default_config.int_val == -1
+    assert default_config.str_val == 'value-1'
+    extend_with_default_external_config(default_config)
+    assert default_config.int_val == 2
+    assert default_config.str_val == 'value2'
+
+
+def test_extend_with_default_external_config_all(default_config, default_config_file_system):
+    assert default_config.int_val == -1
+    assert default_config.str_val == 'value-1'
+    extend_with_default_external_config(default_config)
+    assert default_config.int_val == 3
+    assert default_config.str_val == 'value3'
```

Comment:
Can you please add a test for `getConfig()` as well? (which checks that getConfig() takes all necessary configs)

Reply:
wait for Sergey's fix